### PR TITLE
Amend JSON schema

### PR DESCRIPTION
**Pull request by andersy005**
#27 was merged by accident, and this PR amends it.

### CODE REVIEW

File: esm-collection/json-schema/schema.json [hunk 1]
```diff
@@ -1,14 +1,15 @@
 {
-  "$schema": "http://json-schema.org/draft-07/schema#",
-  "$id": "schema.json#",
   "title": "ESM Extension Specification",
   "description": "This object represents Collections in a Earth System Model Catalog.",
   "allOf": [
     {
-      "$ref": "https://raw.githubusercontent.com/radiantearth/stac-spec/v1.0.0-beta.1/collection-spec/json-schema/collection.json"
+      "$ref": "#/definitions/catalog"
```
Comment (m-mohr) [inline esm-collection/json-schema/schema.json]: Catalog is already included in Collection, so the ref to catalog can be removed.

File: esm-collection/json-schema/schema.json [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
     },
     {
-      "$ref": "https://raw.githubusercontent.com/radiantearth/stac-spec/v1.0.0-beta.1/extensions/collection-assets/json-schema/schema.json"
+      "$ref": "#/definitions/collection"
```
Comment (m-mohr) [inline esm-collection/json-schema/schema.json]: We started hosting them correctly:

```suggestion
      "$ref": "https://schemas.stacspec.org/v1.0.0-beta.2/collection-spec/json-schema/collection.json"
```

File: esm-collection/json-schema/schema.json [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+    },
+    {
+      "$ref": "#/definitions/assets"
```
Comment (m-mohr) [inline esm-collection/json-schema/schema.json]:
```suggestion
      "$ref": "https://schemas.stacspec.org/v1.0.0-beta.2/extensions/collection-assets/json-schema/schema.json"
```

File: esm-collection/json-schema/schema.json [hunk 4]
```diff
@@ -163,6 +168,262 @@
...
+              "title": {
+                "title": "Link title",
+                "type": "string"
+              }
+            }
+          }
+        }
+      }
+    },
+    "collection": {
```
Comment (m-mohr) [inline esm-collection/json-schema/schema.json]: Remove this

File: esm-collection/json-schema/schema.json [hunk 5]
```diff
@@ -163,6 +168,262 @@
...
+          "title": "Asset links",
+          "description": "Links to assets",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/asset"
+          }
+        }
+      }
+    },
+    "catalog": {
```
Comment (m-mohr) [inline esm-collection/json-schema/schema.json]: Remove this

File: esm-collection/json-schema/schema.json [hunk 6]
```diff
@@ -27,16 +28,20 @@
               {
                 "allOf": [
                   {
-                    "$ref": "https://raw.githubusercontent.com/radiantearth/stac-spec/v1.0.0-beta.1/item-spec/json-schema/item.json#/definitions/asset"
+                    "$ref": "#/definitions/asset"
```
Comment (m-mohr) [inline esm-collection/json-schema/schema.json]:
```suggestion
                    "$ref": "https://schemas.stacspec.org/v1.0.0-beta.2/item-spec/json-schema/item.json#/definitions/asset"
```

File: esm-collection/json-schema/schema.json [hunk 7]
```diff
@@ -163,6 +168,262 @@
...
+        "roles": {
+          "title": "Asset roles",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "assets": {
```
Comment (m-mohr) [inline esm-collection/json-schema/schema.json]: Remove this

File: esm-collection/json-schema/schema.json [hunk 8]
```diff
@@ -163,6 +168,262 @@
           }
         }
       }
+    },
+    "asset": {
```
Comment (m-mohr) [inline esm-collection/json-schema/schema.json]: Remove this

**Review by m-mohr — COMMENTED** [review]
We've changed JSON Schemas recently and are now hosting them, so we can simplify a lot here... see comments. There's likely more to do, but it probably easier to clean-up first and then have a look again.

Other than that, I think we should try to get the markdown description ready and then it's easier to make a JSON schema for it :-)

File: esm-collection/README.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+This document explains the fields of the STAC Earth System Model (ESM) Extension to a STAC Collection.
+ESM data is considered to be data produced by simulations models of the earth for a data specific model domain and temporal extent.
+
+- [Example](./examples/sample-pangeo-cmip6-collection.json)
+- [JSON Schema](./json-schema/schema.json)
+
+## Collection Fields
+
+This extension introduces a three new fields at the top level of the collection: `esm:catalog`, `esm:attributes` and `esm:aggregation_control`.
```
Comment (andersy005) [inline esm-collection/README.md]: @m-mohr, I could use your input here :) 

I am trying to figure out whether the following fields belong in this section of the README?

- `type`
- `roles`
- `esm:data_field`
- `esm:format`
- `esm:attributed_field`

Comment (m-mohr) [reply esm-collection/README.md]: Hmm, I guess type and roles doesn't belong here... they have no prefix and are likely for assets.

The others (esm:...) likely belong here, but it was a while ago we did this and I can't fully remember what all these fielda are about.

I need to think into this again, but it's already 10PM here so will postpone that ;-)

Comment (andersy005) [reply esm-collection/README.md]: Thank you for your feedback! 

> I need to think into this again, but it's already 10PM here so will postpone that ;-)

I understand. Are you around tomorrow or next week? I'd like to ping you for additional help on this :)

Comment (m-mohr) [reply esm-collection/README.md]: I'm off work until Tuesday, but I'm looking into mails/Gitter etc every once in a while to help out during the data sprint.
I guess best would be to have a short call tomorrow (can use the data sprint main room) or so to discuss a bit and then you can likely finish it.

Comment (andersy005) [reply esm-collection/README.md]: Awesome! Yes, a short call would be helpful. Let me know what time works best for you tomorrow and we shall schedule something. My schedule is flexible tomorrow.

Comment (m-mohr) [reply esm-collection/README.md]: What timezone are you in? Would 14:00 UTC work for you? But I'm pretty flexible too

Comment (andersy005) [reply esm-collection/README.md]: I am in Mountain Daylight Time zone. 14:00 UTC (8:00 AM local time) works for me

Screen Shot 2020-09-10 at 2 58 35 PM

Comment (m-mohr) [reply esm-collection/README.md]: We can also do 1 or 2 hours later if that's better for you.

Comment (andersy005) [reply esm-collection/README.md]: `14:00 UTC` is a good one (before getting distracted by other tasks later in the day :) )

File: .github/workflows/ci.yaml [hunk 10]
```diff
@@ -25,12 +25,13 @@ jobs:
...
           remark -f -u validate-links .
 
-      - name: Validate sample collections
-        run: |
-          cd collection-spec/examples
-          esmcol-validator sample-glade-cmip6-netcdf-collection.json
-          esmcol-validator sample-pangeo-cmip6-collection.json
-          esmcol-validator --esmcol-spec-dirs ../json-schema simple-collection-with-catalog-dict.json
+
+      # - name: Validate sample collections
```
Comment (m-mohr) [inline .github/workflows/ci.yaml:29]: You can likely replace it with one of these: https://github.com/m-mohr/stac-node-validator/blob/master/COMPARISON.md

File: esm-collection/json-schema/schema.json [hunk 11]
```diff
@@ -1,14 +1,12 @@
 {
-  "$schema": "http://json-schema.org/draft-07/schema#",
-  "$id": "schema.json#",
   "title": "ESM Extension Specification",
   "description": "This object represents Collections in a Earth System Model Catalog.",
   "allOf": [
     {
```
Comment (m-mohr) [inline esm-collection/json-schema/schema.json:7]: I looked at the other extenal extension again: https://github.com/radiantearth/stac-spec/issues/874#issuecomment-675683146

There, I don't refer to the actual specs. The disadvantage with the links/schemas is that they are bound to a specific version and won't work for any future version unless the links are changed here as well.

Maybe the best way is to remove them and add the required collection-assets extension to stac_extensions in the schema. That way validators should handle it correctly.
The issue with the links

File: esm-collection/README.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+
+| Field Name              | Type                                                          | Description                                                                                                                                   |
+| ----------------------- | ------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------- |
+| esm:catalog             | Map<string, [ESM Catalog Object](#esm-catalog-object)>        | A dictionary of assets. Required if `assets.catalog` not defined.                                                                             |
+| esm:attributes          | [string]                                                      | **REQUIRED.** A list of attribute columns in the data set.                                                                                    |
+| esm:aggregation_control | [Aggregation Control Object](#esm-aggregation-control-object) | **OPTIONAL.** Description of how to support aggregation of multiple assets into a single xarray data set.                                     |
```
Comment (m-mohr) [inline esm-collection/README.md]:
```suggestion
| esm:aggregation_control | [Aggregation Control Object](#esm-aggregation-control-object) | Description of how to support aggregation of multiple assets into a single xarray data set.                                     |
```

**Comment by rabernat** [comment]
I'd love to help revive this. Could someone do me the favor of providing a quick summary of where we stand with the esm-collection STAC extension?

**Comment by andersy005** [comment]
> I'd love to help revive this. Could someone do me the favor of providing a quick summary of where we stand with the esm-collection STAC extension?

Thank you for your patience, @rabernat! Regarding the summary, it's my understanding that 80%-90% of the work needed to bring this effort to completion is done, and the critical pieces missing here are:

- formal validation of the schema against STAC
- finalizing the specification document: https://github.com/andersy005/esm-collection-spec/blob/update-stac-spec/esm-collection/README.md

**Comment by andersy005** [comment]
P.S. I'm going to work on pushing this forward this week

**Comment by rabernat** [comment]
I'd suggest we discuss a bit before moving forward. Things are evolving rapidly on CMIP6 cloud stuff. See e.g. discussion at https://github.com/pangeo-forge/cmip6-pipeline/issues/7.

I'm no longer sure we really need esm-collection. I think we can just index our cloud data into a hierarchical STAC catalog. Example here: https://nbviewer.jupyter.org/github/pangeo-forge/cmip6-pipeline/blob/master/catalog_stuff/crawl-serial.ipynb

Once we have such a collection, we could provide search capabilities via the STAC search API.

The CMIP6 cloud data group has been meeting biweekly on Fridays at 10am ET. We are meeting this Friday. @andersy005 I just invited you if you want to join. Your input would be very welcome.

**Comment by andersy005** [comment]
> I'm no longer sure we really need esm-collection. I think we can just index our cloud data into a hierarchical STAC catalog. Example here: nbviewer.jupyter.org/github/pangeo-forge/cmip6-pipeline/blob/master/catalog_stuff/crawl-serial.ipynb

This looks very promising! I was able to run the notebook locally.

> We are meeting this Friday. @andersy005 I just invited you if you want to join. Your input would be very welcome.

Thank you for the invite, @rabernat! I'll be there. In the meantime, I am curious.. 

- What will be the new role of `intake-esm` and/or `intake-stac` under this new approach?
- Will this new approach be flexible enough to support assets in non-cloud native formats such as netCDF/HDF?

> Once we have such a collection, we could provide search capabilities via the STAC search API.

👍  that'd be a very nice feature

**Pull request closed without merging**